hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/HoodieHiveCatalog.java

```diff
@@ -893,7 +893,13 @@ private Map<String, String> supplementOptions(
     } else {
       Map<String, String> newOptions = new HashMap<>(options);
       // set up hive sync options
-      newOptions.put(FlinkOptions.HIVE_SYNC_ENABLED.key(), "true");
+      if (!newOptions.containsKey(FlinkOptions.HIVE_SYNC_ENABLED.key())) {
+        newOptions.put(FlinkOptions.HIVE_SYNC_ENABLED.key(), "true");
+      }
+      if (newOptions.containsKey(FlinkOptions.HIVE_STYLE_PARTITIONING.key())
+          && "true".equals(newOptions.get(FlinkOptions.HIVE_STYLE_PARTITIONING.key()))) {
+        newOptions.put(FlinkOptions.HIVE_SYNC_PARTITION_EXTRACTOR_CLASS_NAME.key(), "org.apache.hudi.hive.HiveStylePartitionValueExtractor");
```

Comment:
Do we need this ? We already so the inference in `HoodieTableFactory`

Reply:
> Do we need this ? We already so the inference in `HoodieTableFactory`

HoodieTableFactory only defines the default value of `FlinkOptions.HIVE_SYNC_PARTITION_EXTRACTOR_CLASS_NAME` if we choose `FlinkOptions.HIVE_STYLE_PARTITIONING` to false and choose `FlinkOptions.HIVE_SYNC_PARTITION_EXTRACTOR_CLASS_NAME` to the default value. Or may be we can inference the relation between `HIVE_STYLE_PARTITIONING` and `HIVE_SYNC_PARTITION_EXTRACTOR_CLASS_NAME` in somewhere like that in `HoodieSyncConfig`?
![1660270438810](https://user-images.githubusercontent.com/46479816/184272458-997a6524-5e0b-43e8-a057-98fc6cf923a1.png)

Reply:
Maybe we should just fix `HoodieTableFactory#setupHiveOptions` first, there is no good way for inference now for flink options yet, was planning to move the inference logic to separate clazz so the other clazz like `HoodieFlinkStreamer` can also use it.

Reply:
There are inferential relationships for some configurations, and it would be great if those relationships could all be identified somewhere. I will close this PR pending this new feature.